how-to-handle-history-mode-files.md

```diff
@@ -0,0 +1,220 @@
+
+NOTE: The `_fivetran_earliest` column shouldn't be added in the destination table. It is introduced to easily identify the earliest record and can be used to optimize the data loads query.
+Below is an example of `replace_file`
+
+Id(PK) | COL1    | _fivetran_start(PK) | _fivetran_end | _fivetran_active | _fivetran_earliest
+---|---------|---------------------| --- |------------------| --- 
+1 | abc     | T1                  | T2-1  | FALSE            | TRUE
+2 | xyz     | T1                  | TMAX | TRUE             | TRUE
+1 | pqr     | T2                  | T3-1 | FALSE            | FALSE
+1 | def | T3                  | TMAX                | TRUE             | FALSE
+
+#### How to Handle Replaces, Updates and Deletes
+
+##### Replace
+
+###### Example 1:
+
+When the `_fivetran_start` column value of destination table is less than  `_fivetran_start` of batch file.
+Suppose the existing table in destination is as below:
+
+Id(PK) | COL1 | COL2 | _fivetran_start(PK) | _fivetran_end | _fivetran_active | _fivetran_synced
+--- |------|----| --- | --- | --- | ---
+1 | abc  | 1  |T1 | T2-1 | FALSE | T100
+1 | pqr | 2  |  T2 | TMAX | TRUE | T101
+2 | mno | 3  | T2  | TMAX | TRUE | T103
+
+At the source, new records are added:
+
+Id(PK) | COL1 | COL2 | Timestamp | Type
+--- | --- | --- |-----------| ---
+1 | def |1 | T3        | Upserted
+1 | ghi | 1 |  T4       | Upserted
+
+The replace batch file will be as follows:
+
+Id(PK) | COL1 | COL2  | _fivetran_start(PK) | _fivetran_end | _fivetran_active | _fivetran_earliest | _fivetran_synced
+--- |------|-------|---------------------| --- | --- | --- | ---
+1 | def  | 1     | T3                  | T4-1 | FALSE | TRUE | T104
+1 | ghi  | 1| T4                  | TMAX | TRUE | FALSE | T105
+
+
+The final destination table will be as follows:
+
+Id(PK) | COL1 | COL2   | _fivetran_start(PK) | _fivetran_end | _fivetran_active | _fivetran_synced
+--- |---|--------|---------------------| --- |------------------| ---
+1  | abc | 1      | T1                  | T2-1 | FALSE            | T100
+1 | pqr | 2      | T2                  | T3-1 | FALSE            | T101
+2  | mno  | 3  | T3                  | TMAX | TRUE             | T103
+1 | def | 1 |T3 | T4-1 | FALSE            | T104
+1  | ghi | 1 | T4 | TMAX | TRUE             | T105
+
+**Explanation:**
+- We got new records for id = 1. 
+- We check for the corresponding earliest record (`_fivetran_earliest` as TRUE), and deleted the existing records from the destination table where the `_fivetran_start` column value in the destination table is greater than or equal to  `_fivetran_start` of batch file.
+- We set `_fivetran_end` of the active record in destination table to `_fivetran_start`-1 of the `_fivetran_earliest` record of batch file.
+- We set the `_fivetran_active` column value for the above updated record to FALSE and `deleted_column` (if present in the destination table) to TRUE.
```

Comment:
Is it really called `deleted_column` in the destination table or something else?

Reply:
In destination table this column name can be anything. This is the column name we will pass to partners